src/components/Table.js

```diff
@@ -0,0 +1,110 @@
+import React, { Component } from 'react'
+import contacts from '../contacts.json'
+import Button from './Button'
+
+class Table extends Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+            allContacts: [...contacts].splice(5),
+            fiveContacts: [...contacts].splice(0, 5)
+        }
+        this.randomContact = this.randomContact.bind(this)
+        this.sortByName = this.sortByName.bind(this)
+        this.sortByPop = this.sortByPop.bind(this)
+        this.deleteContact = this.deleteContact.bind(this)//El valor de this es ignorado cuando la función es llamada con el operador new.
+
+    }
+
+    contactsResult() {
+        return this.state.fiveContacts.map(ele => (
+            <tr key={ele.id}>
+                <td><img className="tablePic" src={ele.pictureUrl} alt='contact img' /></td>
+                <td><p>{ele.name}</p></td>
+                <td><p>{ele.popularity}</p></td>
+                <td><Button actionBut={this.deleteContact} text="Delete" /></td>
+            </tr>
+        ))
+    }
+
+    randomContact() {
+        let index = Math.floor(Math.random() * this.state.allContacts.length);
+        let copyFiveContacts = [...this.state.fiveContacts]
+        let copyAllContacts = [...this.state.allContacts]
+        const elemToPush = copyAllContacts[index]
+        copyFiveContacts.push(elemToPush)
+        copyAllContacts.splice(index, 1)
```

Comment:
`const copyAllContacts = [...this.state.allContacts, this.state.AllContacts[index]]`